src/complex/Utilities/OStreamUtilities.hpp

```diff
@@ -0,0 +1,765 @@
+  bool isBalanced = false;
+
+public:
+  PrintMatrix2D(size_t numrows, size_t numcols)
+  : rows(numrows)
+  , columns(numcols)
+  , matrix(numrows * numcols, std::string("UNINITIALIZED"))
+  {
+  }
+  ~PrintMatrix2D() = default;
+
+  PrintMatrix2D(PrintMatrix2D&) = default;
+  PrintMatrix2D(PrintMatrix2D&&) noexcept = default;
+
+  PrintMatrix2D& operator=(const PrintMatrix2D&) = default;
+  PrintMatrix2D& operator=(PrintMatrix2D&&) noexcept = default;
+
+  std::string& getValue(size_t row, size_t column) // no error checking for walking off edge
+  {
+    return matrix[row * columns + column];
+  }
+
+  std::string& getValue(size_t index) // no error checking for walking off edge
+  {
+    return matrix[index];
+  }
+
+  const std::string& getValue(size_t row, size_t column) const // no error checking for walking off edge
+  {
+    return matrix[row * columns + column];
+  }
+
+  const std::string& getValue(size_t index) const // no error checking for walking off edge
+  {
+    return matrix[index];
+  }
+
+  std::string& operator()(size_t row, size_t column) // no error checking for walking off edge
+  {
+    return matrix[row * columns + column];
+  }
+
+  std::string& operator()(size_t index) // no error checking for walking off edge
+  {
+    return matrix[index];
+  }
+
+  const std::string& operator()(size_t row, size_t column) const // no error checking for walking off edge
+  {
+    return matrix[row * columns + column];
+  }
+
+  const std::string& operator()(size_t index) const // no error checking for walking off edge
+  {
+    return matrix[index];
+  }
+
+  size_t getSize() const
+  {
+    return rows * columns;
+  }
+
+  size_t getRows() const
+  {
+    return rows;
+  }
+  size_t getColumns() const
+  {
+    return columns;
+  }
+  bool getBalance() const
+  {
+    return isBalanced;
+  }
+  /*
+   * This function is used to set balance to true if all values are initialized.
+   */
+  void checkBalanceType() // processor heavy, avoid running if you know its unbalanced
+  {
+    bool balanced = true;
+    for(const auto& value : matrix)
+    {
+      if(value.find("UNINITIALIZED") != std::string::npos)
+      {
+        balanced = false;
+      }
+    }
+    isBalanced = balanced;
+  }
+};
+
+/**
+ * @brief Parallel enabled | implicit loading of **DataArray**'s elements to PrintMatrix2D vertically
+ * @tparam T The primitive type attacthed to **DataArray**
+ * @param outputMatrix The PrintMatrix2D that will store string values from *inputArray*
+ * @param inputArray The **DataArray** that will have its values translated into strings
+ * @param columnToWrite The column that the **DataArray** should be stored in
+ * @param hasHeaders bool to determine if index must be advanced 1 to avoid overwrite
+ */
+template <typename T>
+struct LoadDataArrayToMatrixImpl
+{
+  LoadDataArrayToMatrixImpl(PrintMatrix2D& outputMatrix, const DataArray<T>& inputArray, const size_t& columnToWrite, bool hasHeaders = false)
```

Comment:
```suggestion
  LoadDataArrayToMatrixImpl(PrintMatrix2D& outputMatrix, const DataArray<T>& inputArray, size_t columnToWrite, bool hasHeaders = false)
```
Primitives should be passed by value.